common/OpTestHost.py

```diff
@@ -884,3 +884,46 @@ def host_check_binary(self, i_dir, i_file, console=0):
             l_msg = "Executable file %s/%s is not present" % (i_dir, i_file)
             log.debug(l_msg)
             return False
+
+    ##
+    # @brief Check the Linux kernel log in skiroot and the OS for
+    #        warnings and errors
+    #
+    # @return On Success, None
+    #         On Failure, warning/error message from kernel log
+    #
+    def host_check_kernel_log(self, console=0, filter_out=None):
+        '''
+        Check the Linux kernel log in skiroot and the OS for warnings and errors
+        '''
+
+        log_entries = []
+        # Depending on where we're running, we may need to do all sorts of
+        # things to get a sane dmesg output. Urgh.
+        try:
+            log_entries = self.host_run_command(
+                "dmesg --color=never -T --level=alert,crit,err,warn", console=console)
+        except CommandFailed:
+            try:
+                log_entries = self.host_run_command(
+                    "dmesg -T --level=alert,crit,err,warn", console=console)
+            except CommandFailed:
+                try:
+                    log_entries = self.host_run_command(
+                        "dmesg -r|grep '<[4321]>'", console=console)
+                except CommandFailed as cf:
+                    # An exit code of 1 and no output can mean success.
+                    # as it means we're not successfully grepping out anything
+                    if cf.exitcode == 1 and len(cf.output) == 0:
+                        pass
+
+        if filter_out is not None:
```

Comment:
this is not neeed, below for itself will take care of skipping if none

Reply:
pls provide param description, so that user would know how to provide filter_out, looks like the method expects it as a list but not clear from param or it can be comma separated string and choose to split and use it as list `for f in filter_out.split(','):`